### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by tomdan-ai**
# In This Milestone
- Modify and delete data in SQL.
  - Add a new column to the animals' table
  - Add and delete data from species record

- Use database transactions.
  - Some implemented transactions:
  
Transc1
Transc2
Transc3
Transc4
Transc5
Transc6
Transc7
Transc8

- Prepare complex queries that answer analytical questions.
  - How many animals are there?
   
    Q1

   - How many animals have never tried to escape?
   
     Q2

   - What is the average weight of animals?
      
      Q3

   - Who escapes the most, neutered or not neutered animals?
       
      Q4

   - What is the minimum and maximum weight of each type of animal?

       Q5

   - What is the average number of escape attempts per animal type of those born between 1990 and 2000?

        Q6

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,6 +6,7 @@ CREATE TABLE animals (
   date_of_birth DATE,
   escape_attempts INTEGER default 0,
   neutered BOOLEAN,
-  weight_kg DECIMAL
+  weight_kg DECIMAL,
+  species VARCHAR(100)
```
Comment (Munsa1) [inline schema.sql]: - [ ] The way to document that we are changing the structure of the animal's table is by using the `Alter table` SQL statement. This helps users to see the development of our database. Kindly do so
e.g
`ALTER TABLE animals ADD species varchar(255);`

File: queries.sql [hunk 2]
```diff
@@ -30,3 +30,96 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+BEGIN;
+
+-- Update the species column to "pokemon" for animals without a species
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL OR species = '';
+
+-- Commit the transaction
+COMMIT;
+
+-- Verify the changes
+SELECT * FROM animals;
```
Comment (Munsa1) [inline queries.sql]: - [ ] This code here should be inside one single transaction as per requirements.

Screenshot of requirements
image
e.g
Your codebase should look something like this
```  
 BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

### Note that you should update your screenshots to reflect this change

File: queries.sql [hunk 3]
```diff
@@ -30,3 +30,96 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+
+-- Begin the transaction
+BEGIN;
+
+-- Delete all records from the animals table
+DELETE FROM animals;
+
```
Comment (Munsa1) [inline queries.sql:75]: - [ ] Immediately after deleting all the records we must verify that the change was made using a `select` statement. kindly do so.
e.g
`SELECT COUNT(*) FROM ANIMALS;`

### Note that the screenshot for this transaction should bu updated as well to reflect the change

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @tomdan-ai, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Updated Animals Table :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive Pr :heavy_check_mark: 
Good Commit Message :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (tomdan-ai) [reply queries.sql:75]: Thank you so much for the wonderful suggestions its highly appreciated but I believe the change you request here already exists in my PR in [this](https://user-images.githubusercontent.com/88054834/245828117-c4bd1dab-9f7f-4d38-8f01-6e3ab5d46ae7.jpg) image. 

I would be glad to be given more clarity on this suggestion if I am wrong. Thank you

**Review by Ismailco — APPROVED** [review]
### Hi @tomdan-ai, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**